src/spatial/traversal/IterationDiffAnalyzer.scala

```diff
@@ -0,0 +1,141 @@
+                                          |
+                                          |     II  = lat
+                                                lat = 2 * single lane's latency 
+                                                laneWaits = Map( 0 -> 0, 1 -> 1 )
+            
+                                Foreach(N by 1 par 3){i => mem(i) = mem(i-2)}
+                                  O   O   O   O   O
+                                  |___|___^|  ^   ^
+                                      |____|__|   |
+                                           |______|
+                                                                      
+                   LANE RETIMING           0   1  2                                                 
+                                           |   |                                                 
+                                           |   |                                               
+                                                  |                                                
+                                                  |
+                                                    II  = lat
+                                                    lat = 2 * single lane's latency 
+                                                    laneWaits = Map( 0 -> 0, 1 -> 0, 2 -> 1)
+                  */
+                  val advancePerLane = (thisIterReads(1) - thisIterReads.head).collapse.max
+                  val diffPerLane = (write-read).collapse.max
+                  if (diffPerLane > 0) { // diffPerLane = 0 is a case of mem(i) = mem(i) + ...
+                    val dependsOnLane = List.tabulate(par){i => i * advancePerLane - diffPerLane}
+                    val laneWaitMapping = dependsOnLane.zipWithIndex.map{case(d,i) =>
+                      val laneWait = if (d < 0) 0 else {
+                        ((0 max dependsOnLane(i))/diffPerLane).toInt + 1
+                      }
+                      (i -> laneWait)
+                    }.toMap
+                    dbgs(s"advancePerLane = $advancePerLane, diffPerLane = $diffPerLane, lane dependencies $dependsOnLane")
+                    dbgs(s"laneWaits = ${laneWaitMapping}")
+                    reader.laneWaits = laneWaitMapping
+                    writer.laneWaits = laneWaitMapping
+                    mem.laneWaits = laneWaitMapping
+                    // TODO: Tie this in to unrolling
+                    if (laneWaitMapping.values.toList.distinct.length > 1) throw new Exception("Inter-lane access dependencies in cycles not yet supported")
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+
+ }
+
+  override protected def visit[A](lhs: Sym[A], rhs: Op[A]): Unit = rhs match {
+    case _:AccelScope => inAccel{ super.visit(lhs, rhs) }
+
+    case ctrl: Control[_] => 
+      lhs match {
+        case Op(OpMemReduce(_,_,_,accum,_,_,loadAcc,_,storeAcc,_,_,_,_)) => 
```

Comment:
There is probably a much cleaner way to handle OpMemReduce and OpReduce here.  Finding accum cycles in these before unrolling doesn't work because you can't trace the back-edge through the different pseudostages